source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -84,8 +90,50 @@ void GeoipProviderConfig::incCounter(Stats::StatName name) {
   stats_scope_->counterFromStatName(name).inc();
 }
 
+GeoipProvider::GeoipProvider(Event::Dispatcher& dispatcher, Api::Api& api,
+                             Singleton::InstanceSharedPtr owner,
+                             GeoipProviderConfigSharedPtr config)
+    : config_(config), owner_(owner) {
+  city_db_ =
+      config_->cityDbPath() ? initMaxmindDb(config_->cityDbPath().value(), CITY_DB_TYPE) : nullptr;
+  isp_db_ =
+      config_->ispDbPath() ? initMaxmindDb(config_->ispDbPath().value(), ISP_DB_TYPE) : nullptr;
+  anon_db_ =
+      config_->anonDbPath() ? initMaxmindDb(config_->anonDbPath().value(), ANON_DB_TYPE) : nullptr;
+  mmdb_reload_dispatcher_ = api.allocateDispatcher("mmdb_reload_routine");
```

Comment:
does it make sense to enable/disable the reload feature via a config option or runtime key? seems such option could make it easier to roll out the binary change in prod

Reply:
thank you! yeah, understood it's in alpha, we do run it for some of our prod and has been working well. really looking forward to the hot-reload feature (and safely roll it out)